### PR TITLE
Fixes melee weapon variant inconsistencies.

### PR DESCRIPTION
**Pull request by NewyearnewmeUwu**
#### Summary

SUMMARY: Bugfixes "Fixes melee weapon variant technique differences"

#### Purpose of change

Previously, the fake battleaxe had the wide technique while the legitimate version lacked it. Fake weapons should still break easily, but you should be able to do the same attacks with every version of a melee weapon so long as you possess the skills.

This both bothers my OCD side and also doesn't make logical sense, especially the fake battleaxe possessing the wide attack but not the real one.

#### Describe the solution

I really just went through the weapon JSONs and looked for inconsistencies between fake/inferior/real versions of melee weapons.

#### Describe alternatives you've considered

Ignoring the inconsistencies. 

#### Testing

I copied the modify JSONs over to the latest experimental build (2021-07-11-1759) and all the weapons I changed were properly changed when I spawned them in.

#### Additional context

I have the original fake/inferior weapon code creator's blessing on this, and they were actually the one who suggested all versions of a weapon should have the same techniques.

### CODE REVIEW

**Comment by actual-nh** [comment]
I believe Cal (on discord) is, more precisely, the author of the fake ones, while the inferior ones are someone else's originally.

For the PR validator to work, you will need to move what's currently under "Summary" to under "Purpose of Change", then put in a summary of `Bugfixes "Fixes melee weapon variant inconsistencies"` - or, alternatively, simply "None"; either goes right under the Summary header.

**Comment by somepersondunno** [comment]
> 
> 
> I believe Cal (on discord) is, more precisely, the author of the fake ones, while the inferior ones are someone else's originally.
> 
> For the PR validator to work, you will need to move what's currently under "Summary" to under "Purpose of Change", then put in a summary of `Bugfixes "Fixes melee weapon variant inconsistencies"` - or, alternatively, simply "None"; either goes right under the Summary header.

Been a while, fixed the Summary description. Also just wanted to credit them with the idea because I was originally just gonna fix the battleaxe.

**Comment by actual-nh** [comment]
I don't quite see how adjusting flammable arrows is part of a PR focusing on fake/cheap/good melee weapon variants. Accidental inclusion?

**Comment by somepersondunno** [comment]
> 
> 
> I don't quite see how adjusting flammable arrows is part of a PR focusing on fake/cheap/good melee weapon variants. Accidental inclusion?

Yeah, forgot to make another branch. How do I.. delete that commit?

**Comment by I-am-Erk** [comment]
The ideal way to do this would be to use copy-from so that the fake and inferior weapons inherit from the main one but just have their stats reduced.

**Comment by actual-nh** [comment]
> The ideal way to do this would be to use copy-from so that the fake and inferior weapons inherit from the main one but just have their stats reduced.

For more on this, see doc/JSON_INHERITANCE.md. Flags like DURABLE_MELEE could also be removed (or added for FRAGILE_MELEE).

**Review by Maleclypse — APPROVED** [review]

**Comment by somepersondunno** [comment]
I am a hollow shell of a human being. The sacraments are complete.

File: data/json/items/melee/bludgeons.json [hunk 1]
```diff
@@ -619,27 +592,22 @@
...
+    "copy-from": "lucern_hammer",
     "type": "GENERIC",
-    "symbol": "/",
-    "color": "light_gray",
     "name": { "str": "lucerne hammer" },
-    "description": "This is a cheap piece of crap imitation of a versatile polearm with a spiked hammer head, probably used for LARPing",
-    "price": 4000,
-    "price_postapoc": 500,
+    "description": "This is a cheap piece of crap imitation of a versatile polearm with a spiked hammer head, probably meant to be used for LARPing.",
+    "relative": { "bashing": -19, "cutting": -31 },
```
Comment (I-am-Erk) [inline data/json/items/melee/bludgeons.json:604]:
```suggestion
    "proportional": { "bashing": 0.5, "cutting": 0.1 },
```

File: data/json/items/melee/bludgeons.json [hunk 2]
```diff
@@ -659,46 +627,35 @@
...
+    "category": "weapons"
   },
   {
-    "type": "GENERIC",
     "id": "mace_fake",
+    "copy-from": "mace",
+    "type": "GENERIC",
     "name": { "str": "mace" },
     "description": "A light, cheaply made replica of a medieval weapon that would normally consist of a wood handle with a heavy iron end.",
+    "relative": { "bashing": -25 },
```
Comment (I-am-Erk) [inline data/json/items/melee/bludgeons.json:639]:
```suggestion
    "proportional": { "bashing": 0.38 },
```

File: data/json/items/melee/bludgeons.json [hunk 3]
```diff
@@ -659,46 +627,35 @@
...
+    "copy-from": "mace",
+    "type": "GENERIC",
     "name": { "str": "mace" },
     "description": "A medieval weapon consisting of a wood handle with a heavy iron end.  That iron end feels a bit loose.",
-    "weight": "1000 g",
-    "to_hit": 1,
-    "color": "dark_gray",
-    "symbol": "/",
-    "looks_like": "mace",
+    "relative": { "bashing": -18 },
```
Comment (I-am-Erk) [inline data/json/items/melee/bludgeons.json:653]:
```suggestion
    "proportional": { "bashing": 0.55 },
```

File: data/json/items/melee/bludgeons.json [hunk 4]
```diff
@@ -755,50 +712,37 @@
...
+    "category": "weapons"
   },
   {
-    "type": "GENERIC",
     "id": "morningstar_fake",
+    "copy-from": "morningstar",
+    "type": "GENERIC",
     "name": { "str": "morningstar" },
     "description": "A light, cheaply made replica of a medieval weapon that would normally consist of a wood handle with a heavy, spiked iron ball on the end.",
+    "relative": { "bashing": -25, "cutting": -7 },
```
Comment (I-am-Erk) [inline data/json/items/melee/bludgeons.json:724]:
```suggestion
    "proportional": { "bashing": 0.35, "cutting": 0.1 },
```

File: data/json/items/melee/bludgeons.json [hunk 5]
```diff
@@ -755,50 +712,37 @@
...
   },
   {
-    "type": "GENERIC",
     "id": "morningstar_inferior",
+    "copy-from": "morningstar",
+    "type": "GENERIC",
     "name": { "str": "morningstar" },
-    "description": "A medieval weapon consisting of a wood handle with a heavy, spiked iron ball on the end.  That end feels lighter than it should.",
+    "description": "A light, cheaply made replica of a medieval weapon that would normally consist of a wood handle with a heavy, spiked iron ball on the end.",
+    "relative": { "bashing": -8, "cutting": -7 },
```
Comment (I-am-Erk) [inline data/json/items/melee/bludgeons.json:739]:
```suggestion
    "proportional": { "bashing": 0.7, "cutting": 0.1 },
```

**Review by I-am-Erk — COMMENTED** [review]
I have to run along for a bit, you can continue this if you like or I'll do more later.

**Review by I-am-Erk — COMMENTED** [review]
Another chunk done for today.

We could also just explicitly declare values for remaining ones.

**Comment by somepersondunno** [comment]
Quick in and out! 20 minute PR.

File: data/json/items/melee/bludgeons.json [hunk 6]
```diff
@@ -230,45 +217,37 @@
...
+    "copy-from": "bokken",
     "type": "GENERIC",
     "symbol": "/",
     "color": "brown",
     "looks_like": "bokken",
     "name": { "str_sp": "bokken" },
     "description": "This is a solid wood 'training' katana, but feels far too light to make an effective weapon.",
-    "price": 12500,
-    "price_postapoc": 500,
+    "proportional": { "bashing": 0.5, "cutting": 0 },
```
Comment (I-am-Erk) [inline data/json/items/melee/bludgeons.json:227]:
```suggestion
    "proportional": { "bashing": 0.5 },
    "cutting": 0,
```

**Comment by somepersondunno** [comment]
This should be it, right? Right?!

**Comment by anothersimulacrum** [comment]
I am not sure what is going on with the DPS tests here - values that pass locally fail in the tests here.

File: tests/effective_dps_test.cpp [hunk 7]
```diff
@@ -377,14 +377,14 @@ TEST_CASE( "expected weapon dps", "[expected][dps]" )
         calc_expected_dps( test_guy, "shillelagh", 20.0 );
         calc_expected_dps( test_guy, "bokken", 20.0 );
         calc_expected_dps( test_guy, "PR24-extended", 20.0 );
-        calc_expected_dps( test_guy, "mace_inferior", 18.0 );
+        calc_expected_dps( test_guy, "mace_inferior", 18.5 );
```
Comment (anothersimulacrum) [inline tests/effective_dps_test.cpp:380]:
```suggestion
        calc_expected_dps( test_guy, "mace_inferior", 17.5 );
```

File: tests/effective_dps_test.cpp [hunk 8]
```diff
@@ -377,14 +377,14 @@ TEST_CASE( "expected weapon dps", "[expected][dps]" )
...
         calc_expected_dps( test_guy, "tonfa", 17.0 );
         calc_expected_dps( test_guy, "tonfa_wood", 16.0 );
         calc_expected_dps( test_guy, "shocktonfa_off", 16.0 );
         calc_expected_dps( test_guy, "shocktonfa_on", 16.0 );
         calc_expected_dps( test_guy, "crowbar", 15.0 );
         calc_expected_dps( test_guy, "morningstar_inferior", 15.0 );
-        calc_expected_dps( test_guy, "bokken_inferior", 13.0 );
+        calc_expected_dps( test_guy, "bokken_inferior", 14.0 );
```
Comment (anothersimulacrum) [inline tests/effective_dps_test.cpp:387]:
```suggestion
        calc_expected_dps( test_guy, "bokken_inferior", 13.0 );
```

File: tests/effective_dps_test.cpp [hunk 9]
```diff
@@ -451,12 +451,12 @@ TEST_CASE( "expected weapon dps", "[expected][dps]" )
         calc_expected_dps( test_guy, "machete", 20.0 );
         calc_expected_dps( test_guy, "dao", 18.0 );
         calc_expected_dps( test_guy, "sword_cane", 17.5 );
-        calc_expected_dps( test_guy, "cutlass_inferior", 15.0 );
+        calc_expected_dps( test_guy, "cutlass_inferior", 17.5 );
```
Comment (anothersimulacrum) [inline tests/effective_dps_test.cpp:454]:
```suggestion
        calc_expected_dps( test_guy, "cutlass_inferior", 16.5 );
```

File: tests/effective_dps_test.cpp [hunk 10]
```diff
@@ -451,12 +451,12 @@ TEST_CASE( "expected weapon dps", "[expected][dps]" )
...
         calc_expected_dps( test_guy, "scimitar_inferior", 14.5 );
         calc_expected_dps( test_guy, "sword_crude", 14.0 );
-        calc_expected_dps( test_guy, "wakizashi_inferior", 13.0 );
+        calc_expected_dps( test_guy, "wakizashi_inferior", 15.0 );
```
Comment (anothersimulacrum) [inline tests/effective_dps_test.cpp:457]:
```suggestion
        calc_expected_dps( test_guy, "wakizashi_inferior", 14.0 );
```

File: tests/effective_dps_test.cpp [hunk 11]
```diff
@@ -474,17 +474,17 @@ TEST_CASE( "expected weapon dps", "[expected][dps]" )
         calc_expected_dps( test_guy, "tanto", 18.0 );
         calc_expected_dps( test_guy, "kris", 18.0 );
         calc_expected_dps( test_guy, "knife_rambo", 17.0 );
-        calc_expected_dps( test_guy, "tanto_inferior", 15.0 );
+        calc_expected_dps( test_guy, "tanto_inferior", 16.5 );
```
Comment (anothersimulacrum) [inline tests/effective_dps_test.cpp:477]:
```suggestion
        calc_expected_dps( test_guy, "tanto_inferior", 15.0 );
```

File: tests/effective_dps_test.cpp [hunk 12]
```diff
@@ -474,17 +474,17 @@ TEST_CASE( "expected weapon dps", "[expected][dps]" )
...
-        calc_expected_dps( test_guy, "kirpan_cheap", 11.0 );
+        calc_expected_dps( test_guy, "kirpan_cheap", 10.5 );
         calc_expected_dps( test_guy, "switchblade", 10.5 );
         calc_expected_dps( test_guy, "knife_meat_cleaver", 8.5 );
         calc_expected_dps( test_guy, "diveknife", 8.0 );
         calc_expected_dps( test_guy, "copper_knife", 8.0 );
         calc_expected_dps( test_guy, "knife_butcher", 7.5 );
         calc_expected_dps( test_guy, "throwing_knife", 7.0 );
-        calc_expected_dps( test_guy, "tanto_fake", 7.75 );
+        calc_expected_dps( test_guy, "tanto_fake", 7.5 );
```
Comment (anothersimulacrum) [inline tests/effective_dps_test.cpp:487]:
```suggestion
        calc_expected_dps( test_guy, "tanto_fake", 6.5 );
```

**Review by anothersimulacrum — COMMENTED** [review]
Let's try these.

**Comment by I-am-Erk** [comment]
If these don't work I will revert the proportionals to static statements

**Comment by kevingranade** [comment]
Merge conflict resolved.

File: data/json/items/melee/spears_and_polearms.json [hunk 13]
```diff
@@ -539,50 +523,35 @@
...
+    "flags": [ "DURABLE_MELEE", "POLEARM", "SPEAR", "REACH_ATTACK", "REACH3", "NONCONDUCTIVE", "ALWAYS_TWOHAND" ],
+    "category": "weapons"
   },
   {
     "id": "pike_fake",
+    "copy-from": "pike",
     "type": "GENERIC",
-    "category": "weapons",
-    "name": { "ctxt": "weapon", "str": "pike" },
+    "name": { "str": "pike" },
```
Comment (akirashirosawa) [inline data/json/items/melee/spears_and_polearms.json:533]:
```
- "name": { "ctxt": "weapon", "str": "pike" },
+ "name": { "str": "pike" },
```
Do not remove context, please!

File: data/json/items/melee/spears_and_polearms.json [hunk 14]
```diff
@@ -539,50 +523,35 @@
...
+    "price": 4000,
+    "price_postapoc": 500
   },
   {
     "id": "pike_inferior",
+    "copy-from": "pike",
     "type": "GENERIC",
-    "category": "weapons",
-    "name": { "ctxt": "weapon", "str": "pike" },
+    "name": { "str": "pike" },
```
Comment (akirashirosawa) [inline data/json/items/melee/spears_and_polearms.json:548]:
```
- "name": { "ctxt": "weapon", "str": "pike" },
+ "name": { "str": "pike" },
```
Do not remove context, please!

**Review by akirashirosawa — COMMENTED** [review]
Return the context for the translation.

**Pull request merged**